In [1]:
import pandas as pd
import numpy as np

In [2]:
df= pd.read_csv("D:/python/hitters.csv")

In [3]:
df.head()

Name  AtBat  Hits  HmRun  Runs  RBI  Walks  Years  CAtBat  \
0     Andy Allanson    293    66      1    30   29     14      1     293   
1        Alan Ashby    315    81      7    24   38     39     14    3449   
2       Alvin Davis    479   130     18    66   72     76      3    1624   
3      Andre Dawson    496   141     20    65   78     37     11    5628   
4  Andres Galarraga    321    87     10    39   42     30      2     396   

   CHits  ...  CRuns  CRBI  CWalks  League Division PutOuts  Assists  Errors  \
0     66  ...     30    29      14       A        E     446       33      20   
1    835  ...    321   414     375       N        W     632       43      10   
2    457  ...    224   266     263       A        W     880       82      14   
3   1575  ...    828   838     354       N        E     200       11       3   
4    101  ...     48    46      33       N        E     805       40       4   

   Salary  NewLeague  
0     NaN          A  
1   475.0          N  
2   480.0          A  
3   500.0          N  
4    91.5          N  

[5 rows x 21 columns]

Scenario 1: Find of people who have batted for more than 400 matches

In [10]:
(df[(df['AtBat']>400) ])

Name  AtBat  Hits  HmRun  Runs  RBI  Walks  Years  CAtBat  \
2        Alvin Davis    479   130     18    66   72     76      3    1624   
3       Andre Dawson    496   141     20    65   78     37     11    5628   
5    Alfredo Griffin    594   169      4    74   51     35     11    4408   
9     Andre Thornton    401    92     17    49   66     65     13    5206   
10     Alan Trammell    574   159     21   107   75     59     10    4631   
..               ...    ...   ...    ...   ...  ...    ...    ...     ...   
317     Willie McGee    497   127      7    65   48     37      5    2703   
318  Willie Randolph    492   136      5    76   50     94     12    5511   
319   Wayne Tolleson    475   126      3    61   43     52      6    1700   
320    Willie Upshaw    573   144      9    85   60     78      8    3198   
321    Willie Wilson    631   170      9    77   44     31     11    4908   

     CHits  ...  CRuns  CRBI  CWalks  League Division PutOuts  Assists  \
2      457  ...    224   266     263       A        W     880       82   
3     1575  ...    828   838     354       N        E     200       11   
5     1133  ...    501   336     194       A        W     282      421   
9     1332  ...    784   890     866       A        E       0        0   
10    1300  ...    702   504     488       A        E     238      445   
..     ...  ...    ...   ...     ...     ...      ...     ...      ...   
317    806  ...    379   311     138       N        E     325        9   
318   1511  ...    897   451     875       A        E     313      381   
319    433  ...    217    93     146       A        W      37      113   
320    857  ...    470   420     332       A        E    1314      131   
321   1457  ...    775   357     249       A        W     408        4   

     Errors    Salary  NewLeague  
2        14   480.000          A  
3         3   500.000          N  
5        25   750.000          A  
9         0  1100.000          A  
10       22   517.143          A  
..      ...       ...        ...  
317       3   700.000          N  
318      20   875.000          A  
319       7   385.000          A  
320      12   960.000          A  
321       3  1000.000          A  

[151 rows x 21 columns]

In [13]:
df[df.AtBat >400].Name

2          Alvin Davis
3         Andre Dawson
5      Alfredo Griffin
9       Andre Thornton
10       Alan Trammell
            ...       
317       Willie McGee
318    Willie Randolph
319     Wayne Tolleson
320      Willie Upshaw
321      Willie Wilson
Name: Name, Length: 151, dtype: object

People batted more than 400 matches are 151.

Scenario 2: Find out the name of the player who has hit max home runs

In [19]:
df.sort_values(by=["HmRun"],ascending=False)

Name  AtBat  Hits  HmRun  Runs  RBI  Walks  Years  CAtBat  \
136  Jesse Barfield    589   170     40   107  108     69      6    2325   
80     Dave Kingman    561   118     35    70   94     33     16    6677   
115     Gary Gaetti    596   171     34    91  108     52      6    2862   
140    Jose Canseco    600   144     33    85  117     65      2     696   
244        Rob Deer    466   108     33    75   86     72      3     652   
..              ...    ...   ...    ...   ...  ...    ...    ...     ...   
279     Steve Jeltz    439    96      0    44   36     65      4     711   
64       Doug Baker     24     3      0     1    0      2      3     159   
62   Curt Wilkerson    236    56      0    27   15     11      4    1115   
52    Cliff Johnson     19     7      0     1    2      1      4      41   
250     Ricky Jones     33     6      0     2    4      7      1      33   

     CHits  ...  CRuns  CRBI  CWalks  League Division PutOuts  Assists  \
136    634  ...    371   376     238       A        E     368       20   
80    1575  ...    901  1210     608       A        W     463       32   
115    728  ...    361   401     224       A        W     118      334   
140    173  ...    101   130      69       A        W     319        4   
244    142  ...    102   109     102       A        E     286        8   
..     ...  ...    ...   ...     ...     ...      ...     ...      ...   
279    148  ...     68    56      99       N        E     229      406   
64      28  ...     20    12       9       A        W      80        4   
62     270  ...    116    64      57       A        W     125      199   
52      13  ...      3     4       4       A        E       0        0   
250      6  ...      2     4       7       A        W     205        5   

     Errors  Salary  NewLeague  
136       3  1237.5          A  
80        8     NaN          A  
115      21   900.0          A  
140      14   165.0          A  
244       8   215.0          A  
..      ...     ...        ...  
279      22   150.0          N  
64        0     NaN          A  
62       13   230.0          A  
52        0     NaN          A  
250       4     NaN          A  

[322 rows x 21 columns]

In [20]:
df[df.HmRun==40].iloc[:,:4]

Name  AtBat  Hits  HmRun
136  Jesse Barfield    589   170     40

Scenario 3: Find out league wise total number of players

In [21]:
df.groupby(by=['League']).count().iloc[:,0:1]

Name
League      
A        175
N        147

Scenario 6: Find out division wise total amount of salary invested

In [22]:
df.groupby(by=['Division']).Salary.sum()

Division
E    80531.006
W    60417.501
Name: Salary, dtype: float64

Scenario 10: Find out five most senior players.

In [23]:
(df.groupby(by=['Name']).Years.max().iloc[16:]).sort_values(ascending=False).head()

Name
Pete Rose         24
Tony Perez        23
Graig Nettles     20
Reggie Jackson    20
Ted Simmons       19
Name: Years, dtype: int64

# 2nd df

In [46]:
gog=pd.read_csv("D:/python/Google_Stock_Price_Train.csv")

In [47]:
gog.tail()

Date    Open    High     Low   Close     Volume
1253  12/23/2016  790.90  792.74  787.28  789.91    623,400
1254  12/27/2016  790.68  797.86  787.66  791.55    789,100
1255  12/28/2016  793.70  794.23  783.20  785.05  1,153,800
1256  12/29/2016  783.33  785.93  778.92  782.79    744,300
1257  12/30/2016  782.75  782.78  770.41  771.82  1,770,000

In [48]:
gog.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1258 entries, 0 to 1257
Data columns (total 6 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Date    1258 non-null   object 
 1   Open    1258 non-null   float64
 2   High    1258 non-null   float64
 3   Low     1258 non-null   float64
 4   Close   1258 non-null   object 
 5   Volume  1258 non-null   object 
dtypes: float64(3), object(3)
memory usage: 59.1+ KB


Add 4 new columns to this dataframe, containing open, low, high, closing price in Rupees.(right now stock prices are in dollars)

Close data type is obbect due to  comma  thats why we remove the comma

In [53]:
gog['Close']=gog['Close'].str.replace(',','').astype(float)

In [54]:
gog.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1258 entries, 0 to 1257
Data columns (total 6 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Date    1258 non-null   object 
 1   Open    1258 non-null   float64
 2   High    1258 non-null   float64
 3   Low     1258 non-null   float64
 4   Close   1258 non-null   float64
 5   Volume  1258 non-null   object 
dtypes: float64(4), object(2)
memory usage: 59.1+ KB


In [55]:
new_gog=(gog.assign(openrs =lambda x: x.Open*75,highrs=lambda x: x.High*75,lowrs=lambda x: x.Low*75,closers=lambda x: x.Close*75))

In [56]:
new_gog

Date    Open    High     Low   Close      Volume    openrs  \
0       1/3/2012  325.25  332.83  324.97  663.59   7,380,500  24393.75   
1       1/4/2012  331.27  333.87  329.08  666.45   5,749,400  24845.25   
2       1/5/2012  329.83  330.75  326.89  657.21   6,590,300  24737.25   
3       1/6/2012  328.34  328.77  323.68  648.24   5,405,900  24625.50   
4       1/9/2012  322.04  322.29  309.46  620.76  11,688,800  24153.00   
...          ...     ...     ...     ...     ...         ...       ...   
1253  12/23/2016  790.90  792.74  787.28  789.91     623,400  59317.50   
1254  12/27/2016  790.68  797.86  787.66  791.55     789,100  59301.00   
1255  12/28/2016  793.70  794.23  783.20  785.05   1,153,800  59527.50   
1256  12/29/2016  783.33  785.93  778.92  782.79     744,300  58749.75   
1257  12/30/2016  782.75  782.78  770.41  771.82   1,770,000  58706.25   

        highrs     lowrs   closers  
0     24962.25  24372.75  49769.25  
1     25040.25  24681.00  49983.75  
2     24806.25  24516.75  49290.75  
3     24657.75  24276.00  48618.00  
4     24171.75  23209.50  46557.00  
...        ...       ...       ...  
1253  59455.50  59046.00  59243.25  
1254  59839.50  59074.50  59366.25  
1255  59567.25  58740.00  58878.75  
1256  58944.75  58419.00  58709.25  
1257  58708.50  57780.75  57886.50  

[1258 rows x 10 columns]

# 3rd df

In [57]:
imd=pd.read_csv("D:/python/IMDB-Movie-Data.csv")
imd.head(2)

Rank                    Title                     Genre  \
0     1  Guardians of the Galaxy   Action,Adventure,Sci-Fi   
1     2               Prometheus  Adventure,Mystery,Sci-Fi   

                                         Description      Director  \
0  A group of intergalactic criminals are forced ...    James Gunn   
1  Following clues to the origin of mankind, a te...  Ridley Scott   

                                              Actors  Year  Runtime (Minutes)  \
0  Chris Pratt, Vin Diesel, Bradley Cooper, Zoe S...  2014                121   
1  Noomi Rapace, Logan Marshall-Green, Michael Fa...  2012                124   

   Rating   Votes  Revenue (Millions)  Metascore  
0     8.1  757074              333.13       76.0  
1     7.0  485820              126.46       65.0

Filter out all action movies

In [59]:
(imd[imd['Genre'].str.contains('Action',case=False)])

Rank                           Title                       Genre  \
0       1         Guardians of the Galaxy     Action,Adventure,Sci-Fi   
4       5                   Suicide Squad    Action,Adventure,Fantasy   
5       6                  The Great Wall    Action,Adventure,Fantasy   
8       9              The Lost City of Z  Action,Adventure,Biography   
12     13                       Rogue One     Action,Adventure,Sci-Fi   
..    ...                             ...                         ...   
958   959                  3 Days to Kill       Action,Drama,Thriller   
968   969                         Wrecker      Action,Horror,Thriller   
969   970                 The Lone Ranger    Action,Adventure,Western   
990   991  Underworld: Rise of the Lycans    Action,Adventure,Fantasy   
993   994        Resident Evil: Afterlife     Action,Adventure,Horror   

                                           Description            Director  \
0    A group of intergalactic criminals are forced ...          James Gunn   
4    A secret government agency recruits some of th...          David Ayer   
5    European mercenaries searching for black powde...         Yimou Zhang   
8    A true-life drama, centering on British explor...          James Gray   
12   The Rebel Alliance makes a risky move to steal...      Gareth Edwards   
..                                                 ...                 ...   
958  A dying CIA agent trying to reconnect with his...                 McG   
968  Best friends Emily and Lesley go on a road tri...      Micheal Bafaro   
969  Native American warrior Tonto recounts the unt...      Gore Verbinski   
990  An origins story centered on the centuries-old...  Patrick Tatopoulos   
993  While still out to destroy the evil Umbrella C...  Paul W.S. Anderson   

                                                Actors  Year  \
0    Chris Pratt, Vin Diesel, Bradley Cooper, Zoe S...  2014   
4    Will Smith, Jared Leto, Margot Robbie, Viola D...  2016   
5        Matt Damon, Tian Jing, Willem Dafoe, Andy Lau  2016   
8    Charlie Hunnam, Robert Pattinson, Sienna Mille...  2016   
12   Felicity Jones, Diego Luna, Alan Tudyk, Donnie...  2016   
..                                                 ...   ...   
958  Kevin Costner, Hailee Steinfeld, Connie Nielse...  2014   
968  Anna Hutchison, Andrea Whitburn, Jennifer Koen...  2015   
969  Johnny Depp, Armie Hammer, William Fichtner,To...  2013   
990  Rhona Mitra, Michael Sheen, Bill Nighy, Steven...  2009   
993  Milla Jovovich, Ali Larter, Wentworth Miller,K...  2010   

     Runtime (Minutes)  Rating   Votes  Revenue (Millions)  Metascore  
0                  121     8.1  757074              333.13       76.0  
4                  123     6.2  393727              325.02       40.0  
5                  103     6.1   56036               45.13       42.0  
8                  141     7.1    7188                8.01       78.0  
12                 133     7.9  323118              532.17       65.0  
..                 ...     ...     ...                 ...        ...  
958                117     6.2   73567               30.69       40.0  
968                 83     3.5    1210                 NaN       37.0  
969                150     6.5  190855               89.29        NaN  
990                 92     6.6  129708               45.80       44.0  
993                 97     5.9  140900               60.13       37.0  

[303 rows x 12 columns]

# 4th dataset


In [61]:
cre=pd.read_csv("D:/python/Credit.csv")
cre.head(2)

Unnamed: 0  ID   Income  Limit  Rating  Cards  Age  Education  Gender  \
0           1   1   14.891   3606     283      2   34         11    Male   
1           2   2  106.025   6645     483      3   82         15  Female   

  Student Married  Ethnicity  Balance  
0      No     Yes  Caucasian      333  
1     Yes     Yes      Asian      903

In [62]:
cre['Income'].sort_values()

58      10.354
250     10.363
199     10.403
235     10.503
262     10.588
        ...   
275    163.329
261    180.379
355    180.682
323    182.728
28     186.634
Name: Income, Length: 400, dtype: float64

In [63]:
q =[]
for i in cre['Income']:
    if(i <=50):
        q.append('Poor')
    elif(i <=100):
        q.append('Middle')
    elif(i <=150):
        q.append('Upper_Middle')
    else:
        q.append('Rich_affluent')

In [64]:
cre['class']=q

In [66]:
cre.head(15)

Unnamed: 0  ID   Income  Limit  Rating  Cards  Age  Education  Gender  \
0            1   1   14.891   3606     283      2   34         11    Male   
1            2   2  106.025   6645     483      3   82         15  Female   
2            3   3  104.593   7075     514      4   71         11    Male   
3            4   4  148.924   9504     681      3   36         11  Female   
4            5   5   55.882   4897     357      2   68         16    Male   
5            6   6   80.180   8047     569      4   77         10    Male   
6            7   7   20.996   3388     259      2   37         12  Female   
7            8   8   71.408   7114     512      2   87          9    Male   
8            9   9   15.125   3300     266      5   66         13  Female   
9           10  10   71.061   6819     491      3   41         19  Female   
10          11  11   63.095   8117     589      4   30         14    Male   
11          12  12   15.045   1311     138      3   64         16    Male   
12          13  13   80.616   5308     394      1   57          7  Female   
13          14  14   43.682   6922     511      1   49          9    Male   
14          15  15   19.144   3291     269      2   75         13  Female   

   Student Married         Ethnicity  Balance         class  
0       No     Yes         Caucasian      333          Poor  
1      Yes     Yes             Asian      903  Upper_Middle  
2       No      No             Asian      580  Upper_Middle  
3       No      No             Asian      964  Upper_Middle  
4       No     Yes         Caucasian      331        Middle  
5       No      No         Caucasian     1151        Middle  
6       No      No  African American      203          Poor  
7       No      No             Asian      872        Middle  
8       No      No         Caucasian      279          Poor  
9      Yes     Yes  African American     1350        Middle  
10      No     Yes         Caucasian     1407        Middle  
11      No      No         Caucasian        0          Poor  
12      No     Yes             Asian      204        Middle  
13      No     Yes         Caucasian     1081          Poor  
14      No      No  African American      148          Poor

# 5th Dataset

In [68]:
md=pd.read_csv("D:/python/mcdonalds.csv")
md.head()

Date        Item  Quantity
0  12-Jun-2020  Aloo tikki         2
1  12-Jun-2020    McVeggie         1
2  12-Jun-2020        Chai         2
3  12-Jun-2020    McVeggie         1
4  12-Jun-2020       Fries         2

Find out date wise total number of aloo tikkis sold.

In [69]:
m=md[md.Item=='Aloo tikki']

In [70]:
m.groupby(by=['Date'])['Quantity'].sum()

Date
12-Jun-2020     2
13-Jun-2020    18
Name: Quantity, dtype: int64